### PR TITLE
Enable linux build

### PR DESCRIPTION
**Pull request by nohwnd**
## Description
Build on linux using `build.sh`.

### CODE REVIEW

File: test/vstest.console.UnitTests/vstest.console.UnitTests.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
       <FromP2P>true</FromP2P>
     </ProjectReference>
   </ItemGroup>
-  <ItemGroup>
+  <ItemGroup Condition=" '$(OS)' == 'Windows_NT' OR '$(TargetFramework)' != 'net451' ">
```
Comment (nohwnd) [inline test/vstest.console.UnitTests/vstest.console.UnitTests.csproj:22]: This is bleh, but when I just enable it it succeeds on windows but not on linux, when I make it conditional to just NOT net451 it fails on windows. And without the ref it fails on Linux...

**Review by cvpoienaru — APPROVED** [review]

**Review by Haplois — APPROVED** [review]
LGTM.

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj [hunk 2]
```diff
@@ -6,7 +6,8 @@
   <Import Project="$(TestPlatformRoot)scripts/build/TestPlatform.Settings.targets" />
   <PropertyGroup>
     <AssemblyName>Microsoft.VisualStudio.TestPlatform.ObjectModel</AssemblyName>
-    <TargetFrameworks>net45;net451;netstandard2.0;netstandard1.3;netstandard1.0;uap10.0</TargetFrameworks>
+    <TargetFrameworks>net45;net451;netstandard2.0;netstandard1.3</TargetFrameworks>
+    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(TargetFrameworks);uap10.0;netstandard1.0</TargetFrameworks>
```
Comment (Haplois) [inline src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj:10]: Any reason we are skipping NS1.0 build on Linux, is it broken like UAP?

Comment (nohwnd) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj:10]: It seemed so.

**Pull request merged**